site/src/modules/resources/AgentDevcontainerMoreActions.tsx

```diff
@@ -0,0 +1,136 @@
+import { API } from "api/api";
+import { workspaceAgentContainersKey } from "api/queries/workspaces";
+import type {
+	WorkspaceAgent,
+	WorkspaceAgentDevcontainer,
+	WorkspaceAgentListContainersResponse,
+} from "api/typesGenerated";
+import { Button } from "components/Button/Button";
+import { ConfirmDialog } from "components/Dialogs/ConfirmDialog/ConfirmDialog";
+import {
+	DropdownMenu,
+	DropdownMenuContent,
+	DropdownMenuItem,
+	DropdownMenuTrigger,
+} from "components/DropdownMenu/DropdownMenu";
+import { EllipsisVertical } from "lucide-react";
+import { type FC, useId, useState } from "react";
+import { useMutation, useQueryClient } from "react-query";
+
+type AgentDevcontainerMoreActionsProps = {
+	parentAgent: WorkspaceAgent;
+	devcontainer: WorkspaceAgentDevcontainer;
+};
+
+export const AgentDevcontainerMoreActions: FC<
+	AgentDevcontainerMoreActionsProps
+> = ({ parentAgent, devcontainer }) => {
+	const queryClient = useQueryClient();
+	const [isConfirmingDelete, setIsConfirmingDelete] = useState(false);
+	const [open, setOpen] = useState(false);
+	const menuContentId = useId();
+
+	const queryKey = workspaceAgentContainersKey(parentAgent.id);
+
+	const deleteDevContainerMutation = useMutation({
+		mutationFn: async () => {
+			await API.deleteDevContainer({
+				parentAgentId: parentAgent.id,
+				devcontainerId: devcontainer.id,
+			});
+		},
+		onMutate: async () => {
+			await queryClient.cancelQueries({ queryKey });
+			const previousData = queryClient.getQueryData(queryKey);
+
+			queryClient.setQueryData(
+				queryKey,
+				(oldData?: WorkspaceAgentListContainersResponse) => {
+					if (!oldData?.devcontainers) {
+						return oldData;
+					}
+
+					return {
+						...oldData,
+						devcontainers: oldData.devcontainers.map((dc) => {
+							if (dc.id === devcontainer.id) {
+								return {
+									...dc,
+									status: "stopping",
+									container: undefined,
+								};
+							}
+							return dc;
+						}),
+					};
+				},
+			);
+
+			return { previousData };
+		},
+		onError: (_, __, context) => {
```

Comment:
nit: rather than `__` I usually would do something like `_a` and `_b` to make it clearer what the ignored arguments are. not really a hard rule tho, just personal taste.